crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -151,18 +151,43 @@ impl Format<PyFormatContext<'_>> for FormatTrailingComments<'_> {
                 write!(
                     f,
                     [
-                        line_suffix(&format_args![
-                            empty_lines(lines_before_comment),
-                            format_comment(trailing)
-                        ]),
+                        line_suffix(
+                            &format_args![
+                                empty_lines(lines_before_comment),
+                                format_comment(trailing)
+                            ],
+                            0
+                        ),
                         expand_parent()
                     ]
                 )?;
             } else {
+                // TODO(cnpryer): Reserved width should be used when breaking is expected. Breaking may not be
+                //   expected when a trailing end of line comment is not at the end of a collapsed node.
+                // ```python
+                // def fn():
+                //     return np.divide(
+                //         view.variance,
+                //         view.sum_of_weights,
+                //         out=np.full(view.sum_of_weights.shape, np.nan),
+                //         where=view.sum_of_weights**2 > view.sum_of_weights_squared,  # This shouldn't force a break
+                //     )
+                // ```
+                //
+                // ```
+                // A trailing comment at the end of a line has a reserved width to consider during line measurement.
+                // ```python
+                // tup = (
+                //     "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+                // )  # Some comment
+                // ```
                 write!(
                     f,
                     [
-                        line_suffix(&format_args![space(), space(), format_comment(trailing)]),
+                        line_suffix(
+                            &format_args![space(), space(), format_comment(trailing)],
+                            trailing.slice().range().len().into()
```

Comment:
Will update this to dup the width calculation to include configured tab width.

Reply:
Duplicated what should be enough for the relevant text in 686f3da95e45326e45ac2237b0e6322c3c89f78a.